pkg/querier/querier.go

```diff
@@ -190,6 +196,16 @@ func New(cfg Config, limits *validation.Overrides, distributor Distributor, quer
 
 	opts, mqeOpts := engine.NewPromQLEngineOptions(cfg.EngineConfig, tracker, logger, reg)
 
+	if cfg.EngineConfig.MimirQueryEngine.InstantQuerySplitting.Enabled {
+		intermediateCache, err := cache.NewResultsCache(cfg.EngineConfig.MimirQueryEngine.InstantQuerySplitting.IntermediateResultsCache, logger, reg)
+		if err != nil {
+			return nil, nil, nil, nil, fmt.Errorf("failed to init query splitting cache, err: %w", err)
+		} else {
+			mqeOpts.IntermediateResultCache = intermediateCache
+			util_log.Logger.Log("msg", "intermediate results cache enabled", "backend", cfg.EngineConfig.MimirQueryEngine.InstantQuerySplitting.IntermediateResultsCache.Backend)
+		}
+	}
+
```

Comment:
Any reason to do this here rather than than in `pkg/streamingpromql`? Then the same logic will be applied if this config is used in query-frontends.